cmd/process-agent/config_test.go

```diff
@@ -0,0 +1,111 @@
+package main
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	sysconfig "github.com/DataDog/datadog-agent/cmd/system-probe/config"
+	"github.com/DataDog/datadog-agent/pkg/config"
+	oconfig "github.com/DataDog/datadog-agent/pkg/orchestrator/config"
+	"github.com/DataDog/datadog-agent/pkg/process/checks"
+)
+
+func TestProcessDiscovery(t *testing.T) {
+	scfg, ocfg := &sysconfig.Config{}, &oconfig.OrchestratorConfig{}
+
+	cfg := config.Mock()
+
+	// Make sure the process_discovery check can be enabled
+	cfg.Set("process_config.process_discovery.enabled", true)
+	enabledChecks := getChecks(scfg, ocfg, false)
+	assert.Contains(t, enabledChecks, checks.ProcessDiscovery)
+
+	// Make sure the process_discovery check can be disabled
+	cfg.Set("process_config.process_discovery.enabled", false)
+	enabledChecks = getChecks(scfg, ocfg, true)
+	assert.NotContains(t, enabledChecks, checks.ProcessDiscovery)
+
+	// Make sure the process and process_discovery checks are mutually exclusive
+	cfg.Set("process_config.process_discovery.enabled", true)
+	cfg.Set("process_config.process_collection.enabled", true)
+	enabledChecks = getChecks(scfg, ocfg, true)
+	assert.NotContains(t, enabledChecks, checks.ProcessDiscovery)
+}
+
+func TestContainerCheck(t *testing.T) {
+	scfg, ocfg := &sysconfig.Config{}, &oconfig.OrchestratorConfig{}
+
+	cfg := config.Mock()
+
+	// Make sure the container check can be enabled if the process check is disabled
+	cfg.Set("process_config.process_collection.enabled", false)
+	cfg.Set("process_config.container_collection.enabled", true)
+	enabledChecks := getChecks(scfg, ocfg, true)
+	assert.Contains(t, enabledChecks, checks.Container)
+	assert.Contains(t, enabledChecks, checks.RTContainer)
+	assert.NotContains(t, enabledChecks, checks.Process)
+
+	// Make sure that disabling RT disables the rt container check
+	cfg.Set("process_config.disable_realtime_checks", true)
+	enabledChecks = getChecks(scfg, ocfg, true)
+	assert.Contains(t, enabledChecks, checks.Container)
+	assert.NotContains(t, enabledChecks, checks.RTContainer)
+
+	// Make sure the container check cannot be enabled if we cannot access containers
+	enabledChecks = getChecks(scfg, ocfg, false)
+	assert.NotContains(t, enabledChecks, checks.Container)
```

Comment:
nit: we should probably check that `checks.RTContainer` is not enabled.

Reply:
Fixed :)